rabbitmq/resource_queue.go

```diff
@@ -132,12 +134,28 @@ func ReadQueue(d *schema.ResourceData, meta interface{}) error {
 	e := make(map[string]interface{})
 	e["durable"] = queueSettings.Durable
 	e["auto_delete"] = queueSettings.AutoDelete
-	e["arguments"] = queueSettings.Arguments
-	queue[0] = e
 
-	d.Set("settings", queue)
+	// The user may have used either `arguments` or `arguments_json` to populate this originally.
+	// We need to preserve that decision here so that a subsequent Terraform plan for the
+	// same configuration wouldn't produce an errant diff that moves the value from one
+	// to the other without changing any values.
+	// These two arguments are mutually exclusive due to ConflictsWith in the schema.
+	// `arguments` cannot receive any values other than a string (d.Set will fail), therefore any drift
+	// containing nonstring values AND the configuration originated from `arguments`,
+	// will now be encoded to `arguments_json`.
+	if _, ok := d.GetOk("settings.0.arguments_json"); ok || nonStringInArguments(queueSettings.Arguments) {
+		bytes, err := json.Marshal(queueSettings.Arguments)
+		if err != nil {
+			return err
+		}
+		e["arguments_json"] = string(bytes)
+	} else {
+		e["arguments"] = queueSettings.Arguments
+	}
+
+	queue[0] = e
```

Comment:
Nitpick: Do you mind moving the `queue` allocation here above this assignment, just so it's more/immediately obvious there's no risk in addressing the index like this as it is already allocated?